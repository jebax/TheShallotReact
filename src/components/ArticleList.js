import React, { Component } from 'react'
import axios from 'axios'
import { config } from '../config.js'
import ArticlePreview from './ArticlePreview'

export default class ArticleList extends Component {
  state = {
    articles: []
  }

  componentDidMount() {
    axios.get(config.guardianUrl).then(response => {
      console.log(response)
      this.setState(prevState => ({
        articles: response.data.response.results
      }))
    })
  }

  displayPreview = (article, index) => {
    return (
      <article
        key={index}
      >
        <ArticlePreview
          id={index}
          imageUrl={article.fields.thumbnail}
          headline={article.webTitle}
          url={article.webUrl}
        />
      </article>
    )
  }

  render() {
    return (
      <div
        id='article-list'
      >
        {this.state.articles.map((article, index) => {
          return this.displayPreview(article, index)
        })}
      </div>
    )
  }
}
