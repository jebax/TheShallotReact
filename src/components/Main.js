import React, { Component } from 'react'
import ArticleList from './ArticleList'
import ArticleSummary from './ArticleSummary'
import Title from './Title'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'

export default class Main extends Component {
  render() {
    return (
      <Router>
        <main>
          <Title />
          <Switch>
            <Route exact path='/' component={ArticleList} />
            <Route exact path='/articles/:id' component={ArticleSummary} />
          </Switch>
        </main>
      </Router>
    )
  }
}
