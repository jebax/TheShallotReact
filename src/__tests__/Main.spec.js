import React from 'react'
import { shallow } from 'enzyme'
import Main from '../components/Main'
import ArticleList from '../components/ArticleList'

describe('<Main />', () => {
  var wrapper

  beforeAll(() => {
    wrapper = shallow(<Main />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the Title', () => {
    expect(wrapper.find('Title').length).toEqual(1)
  })

  it('renders the main element', () => {
    expect(wrapper.find('main').length).toEqual(1)
  })

  it('renders two Routes', () => {
    expect(wrapper.find('Route').length).toEqual(2)
  })
})
