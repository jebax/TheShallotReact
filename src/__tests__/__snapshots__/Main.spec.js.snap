// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <main>
        <Title />
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/articles/:id"
          />
        </Switch>
      </main>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Title />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/articles/:id"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/articles/:id"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/articles/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "main",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <main>
          <Title />
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/articles/:id"
            />
          </Switch>
        </main>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Title />,
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/articles/:id"
              />
            </Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/articles/:id"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/articles/:id",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "main",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
